src/applications/vaos/actions/newAppointment.js

```diff
@@ -196,6 +202,51 @@ export function openClinicPage(page, uiSchema, schema) {
   };
 }
 
+export function openSelectAppointmentPage(page, uiSchema, schema) {
+  return async (dispatch, getState) => {
+    let slots;
+    let mappedSlots = [];
+
+    dispatch({
+      type: FORM_SCHEDULE_APPOINTMENT_PAGE_OPENED,
+    });
+
+    try {
+      const response = await getAvailableSlots(
+        getState().newAppointment.data.clinicId,
+      );
+
+      slots = response[0]?.appointmentTimeSlot;
+
+      const now = moment();
+
+      for (let index = 0; index < slots.length; index++) {
```

Comment:
Have read that for loop is still fastest when dealing with smaller sample sizes, but performance is probably negligible here. Replaced with reduce since it _does_ look better :)